src/home/NotificationCenter.tsx

```diff
@@ -19,9 +24,114 @@ import useSelector from 'src/redux/useSelector'
 import colors from 'src/styles/colors'
 import fontStyles from 'src/styles/fonts'
 import { Spacing } from 'src/styles/styles'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'NotificationCenter'
 
 type NotificationsProps = NativeStackScreenProps<StackParamList, Screens.NotificationCenter>
 
+function useCleverTapNotifications() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const messages = useSelector(cleverTapInboxMessagesSelector)
+
+  const notifications: Notification[] = []
+
+  if (messages) {
+    for (const message of messages) {
+      const messageId = message.id ?? message._id
```

Comment:
could we do all this data massaging in the selector with `createSelector` instead? we'd get memoization for free then :)